processor/logdeduplicationprocessor/factory.go

```diff
@@ -0,0 +1,51 @@
+// Copyright  observIQ, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package logdeduplicationprocessor
+
+import (
+	"context"
+	"fmt"
+
+	"go.opentelemetry.io/collector/component"
+	"go.opentelemetry.io/collector/consumer"
+	"go.opentelemetry.io/collector/processor"
+)
+
+const (
+	// typeStr is the value of the "type" key in configuration.
+	typeStr = "logdeduplication"
```

Comment:
Is there a shorter name we could use for this? For instance, could we use just dedup? This might open it up in the future if there's ever an argument to add capabilities for metrics and traces.